pkg/rollout/rollout.go

```diff
@@ -238,12 +280,37 @@ func (r *Rollout) updateAnnotations(svc *run.Service, stable, candidate string)
 		return svc
 	}
 
+	if r.requireRollback {
+		delete(svc.Metadata.Annotations, CandidateRevisionAnnotation)
+		svc.Metadata.Annotations[LastFailedCandidateRevisionAnnotation] = candidate
+	} else {
+		svc.Metadata.Annotations[CandidateRevisionAnnotation] = candidate
+	}
+
```

Comment:
you look at diagnostics and make a decision by changing in-memory SVC object.
then you actually make API call in the same service.
if you split, it's easier to read and test. (also this method no longer reflects "UpdateService")